eldritch/test_events.py

```diff
@@ -1167,6 +1167,199 @@ def testCastAndGoInsane(self):
 
 # TODO: add tests for going unconscious/insane during a mythos/encounter.
 
+class PurchaseTest(EventTest):
+  def testPurchaseOneAtList(self):
+    buy = Purchase(self.char, "common", 1)
+    self.char.dollars = 3
+    self.assertFalse(buy.is_resolved())
+    self.assertFalse(self.char.possessions)
+    food = items.Food()
+    self.state.common.append(food)
+
+    self.state.event_stack.append(buy)
+    choice = self.resolve_to_choice(CardChoice)
+    self.assertEqual(choice.choices, ["Food", "Nothing"])
+    choice.resolve(self.state, "Food")
+    self.resolve_until_done()
+
+    self.assertTrue(buy.is_resolved())
+    self.assertEqual(self.char.dollars, 2)
+    self.assertEqual(self.char.possessions, [food])
+    self.assertFalse(self.state.common)
+
+  def testPurchaseOneAtListDecline(self):
+    buy = Purchase(self.char, "common", 1)
+    self.char.dollars = 3
+    self.assertFalse(buy.is_resolved())
+    self.assertFalse(self.char.possessions)
+    food = items.Food()
+    self.state.common.append(food)
+
+    self.state.event_stack.append(buy)
+    choice = self.resolve_to_choice(CardChoice)
+    self.assertEqual(choice.choices, ["Food", "Nothing"])
+    choice.resolve(self.state, "Nothing")
+    self.resolve_until_done()
+
+    self.assertTrue(buy.is_resolved())
+    self.assertEqual(self.char.dollars, 3)
+    self.assertFalse(self.char.possessions, [])
+    self.assertEqual(len(self.state.common), 1)
+    self.assertEqual(self.state.common[0], food)
+
+  def testPurchaseOneAtListPoor(self):
+    draw = Purchase(self.char, "common", 1)
```

Comment:
Nit: please name this variable `buy` instead of `draw` both to indicate its intent and to match all the other tests.